fixed_bed_model.py

```diff
@@ -0,0 +1,2392 @@
+                m.C[j,'CO2',z,i] = c_CO2(z_,t_)[0]
+
+                m.dCdt[j,'N2',z,i] = dcdt_N2_(z_, t_)[0]
+                m.dCdt[j,'CO2',z,i] = dcdt_CO2_(z_, t_)[0]
+
+                m.v[j,z,i] = V(z_,t_)[0]    
+                m.P[j,z,i] = pres(z_,t_)[0]
+                m.total_den[j,z,i] = tot_den(z_,t_)[0]
+                
+                if m.energy:
+                    m.temp[j,z,i] = temp_e(z_, t_)[0]
+                    m.dTdt[j,z,i] = dTdt_e(z_, t_)[0]
+                    m.nplin[j,z,i] = nplin_e(z_, t_)[0]
+                    
+                if m.isotherm:
+                    m.spp[j,'CO2',z,i] = solid_pres(z_, t_)[0]
+                    x = alpha_scale*(solid_pres(z_, t_)[0] - plin)
+
+                    if alpha_option == 1:
+                        alpha_ = 0.5*(1 - x / np.sqrt(eps_alpha + x*x))
+                    elif alpha_option == 2:
+                        alpha_ = 1 - 1 /(1 + pyo.exp(-x))
+                    elif alpha_option == 3: 
+                        alpha_ = 0.0
+                    elif alpha_option == 4: 
+                        alpha_ = 1.0
+
+                    if alpha_variable:
+                        m.alpha[j,'CO2',z,i] = alpha_
+
+                if m.chemsorb:
+                    m.nchemstar[j,'CO2',z,i] = Nchem_e(z_, t_)[0]
+                    m.dnchemdt[j,'CO2',z,i] = dndt_chem(z_, t_)[0]
+                    m.nchem[j,'CO2',z,i] = Nchem(z_, t_)[0]
+                    if m.energy:
+                        m.nchemstar_mod[j,'CO2',z,i] = alpha_*pyo.value(m.nplin[j,z,i])*solid_pres(z_, t_)[0]/plin + (1-alpha_)*Nchem_e(z_, t_)[0]
+                    else:
+                        m.nchemstar_mod[j,'CO2',z,i] = alpha_*m.nplin*solid_pres(z_, t_)[0]/plin + (1-alpha_)*Nchem_e(z_, t_)[0]
+
+                if m.physsorb:
+                    m.nphysstar[j,'CO2',z,i] = Nphys_e(z_, t_)[0]
+                    #m.nphysstar_mod[j,'CO2',z,i] = alpha_*m.nplin*solid_pres(z_, t_)[0]/plin + (1-alpha_)*Nphys_e(z_, t_)[0]
+                    m.dnphysdt[j,'CO2',z,i] = dndt_phys(z_, t_)[0]
+                    m.nphys[j,'CO2',z,i] = Nphys(z_, t_)[0]   
+            
+            
+
+
+def compute_Kp(mod,LOUD=True):
+    ''' Compute Kp from the solution [m^2/s]
+    '''
+    Kp_ = {}
+    for c in mod.SCOMPS:
+        tfc = pyo.value(mod.fitted_transport_coefficient[c])
+
+        Kp_[c] = radp*radp/15/ads_epsp/tfc
+        
+        if LOUD:
+            print("fitted_transport_coefficient[",c,"] =",tfc,"s")
+            print("Kp[",c,"] =",Kp_[c],"m^2/s")
+        
+    return Kp_
+
+
+### DoE analysis function
+def ext_pres_double(total, option1, option2, node):
+    '''
+    Extract the solution of a variable
+    Arguments:
+        total: a pandas dataframe of solutions
+        option: name of the variable
+        node: number of timesteps
+    Note: the Ngrid is fixed to 20 
+    Return: a list
+    '''
+    x_py1 = total[option1]
+    x_pyomo1 = np.zeros((20,node))
```

Comment:
Why is 20 hard coded? Consider moving to function definition as an optional input.